src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -101,11 +101,16 @@ private[internal] trait TypeMaps {
       case tr @ TypeRef(pre, sym, args) =>
         val pre1 = this(pre)
         val args1 = (
-          if (trackVariance && args.nonEmpty && !variance.isInvariant && sym.typeParams.nonEmpty)
-            mapOverArgs(args, sym.typeParams)
-          else
+          if (trackVariance && args.nonEmpty && !variance.isInvariant) {
+            val tparams = sym.typeParams
```

Comment:
minor: to make it more obvious what happened to the `sym.typeParams.nonEmpty` part of the `if`, this could probably be simplified to
```scala
if (sym.typeParams.nonEmpty) mapOverArgs(args, sym.typeParams)
else args mapConserve this
```